### DISCUSSION TITLE
Static Site, SSR and Hydration

### DISCUSSION THREAD

**Discussion by marcodeluca** [nuxt/nuxt#23758] [Questions] [answered]
Having issues generating a static site, struggling with the concept or value of hydration and if/how I disable it. To summarize the issue:

I'm using Strapi to generate content for a website. Using the latest version of Nuxt3, we are generating all of our routes and pages dynamically. Strapi has the models/content, using npx nuxi generate Nuxt gets the models and relationships, builds our routes and generates our static files.

That all works fine, we end up with all of our folders/files correctly. I serve it up as a static website, but when you navigate to the page, the static content flashes for a half a second and then I guess the page "hydrates" with dynamic data (which doesnt work we get 500 server errors) and the page shows no content. Our issue is pretty much identical to the one that is well documented here in terms of the issue (we do not use storyblok and the proposed solutions do not resolve the issue):

https://github.com/nuxt/nuxt/issues/12377

Each .html page has a couple scripts injected at the bottom of the file that causes the issue. For example, I have a layout called silo.vue that gets our data from Strapi and generates our silo (top level) folders and files. After generating, we'll have a /silo-slug/index.html that loads with our content from strapi for an instant, but then there's some sort of "redirect" back to the dynamic page (/pages/silo/index.vue), trying to access data from strapi which works fine when we're generating static pages but fails now when loading data dynamically. Why it fails "live" is a separate issue though from the main question, what if i just want a static site?

If I manually remove the scripts that nuxt included in every single one of our static files, this no longer happens and the static site works as expected. It just doesnt seem like I can do this via a config? The .output/public folder has a _nuxt folder, inside has a bunch of stuff. So static html files load stuff like:

```
<link rel="modulepreload" as="script" crossorigin href="/_nuxt/entry.782195df.js">
<link rel="modulepreload" as="script" crossorigin href="/_nuxt/index.3425762a.js">
```

The "entry" has the redirect, so this is all the stuff that hijacks my static site:)

I'd like to be able to disable it of course but in general I'm trying to understand why, as a default, I would want to hydrate a static site with dynamic content as a default. I get the concept of wanting the interactivity of vue for dynamic interfaces and pages transitions etc. Im very confused by the implementation. If I'm using Nuxt because i want to generate a static site, assuming I want it to hydrate back into a reactive vue site, I've already got the data for each page during the generate command. Why am I now loading vue pages that are requesting the data all over again?

I havent used Nuxt since it first came out, Nuxt3 seems quite a bit different so maybe it's going in a direction that isn't suitable for generating static sites from dynamic content?

**Comment by warflash** [comment, marked as answer]
This is quite a long question - sorry if I missed anything.
> I'd like to be able to disable it of course but in general I'm trying to understand why, as a default, I would want to hydrate a static site with dynamic content as a default.

When prerendering it should not do fetch requests to the dynamic data you passed. Wrapping that in useAsyncData during generate time should do the trick like the issue mentions. If you can't get that to work and can make a reproducible example then I'd recommend opening an issue to investiage.
The reason why a static site might want to hydrate is like you mentioned the benefits of things like client side routing etc.
If you want no vue/nuxt js shipped with your app which is what I get from the message then try looking into: https://nuxt.com/docs/guide/going-further/experimental-features#noscripts

**Reply by marcodeluca** [reply]
@warflash thanks for the reply. We were able to get the site working once we used useAsyncData properly. I will also look into the noscripts option as well.
